File: app-server/src/main/java/com/github/arhor/simple/expense/tracker/service/ExpenseService.java
package com.github.arhor.simple.expense.tracker.service;

import java.time.LocalDate;
import java.util.List;

import com.github.arhor.simple.expense.tracker.model.ExpenseRequestDTO;
import com.github.arhor.simple.expense.tracker.model.ExpenseResponseDTO;
import com.github.arhor.simple.expense.tracker.util.TemporalRange;

public interface ExpenseService {

    List<ExpenseResponseDTO> getUserExpenses(Long userId, TemporalRange<LocalDate> dateRange);

    ExpenseResponseDTO getExpenseById(Long expenseId, TemporalRange<LocalDate> dateRange);

    ExpenseResponseDTO createUserExpense(Long userId, ExpenseRequestDTO requestDTO);
}

File: app-server/src/main/kotlin/com/github/arhor/simple/expense/tracker/service/ExpenseService.kt
package com.github.arhor.simple.expense.tracker.service

import com.github.arhor.simple.expense.tracker.model.ExpenseRequestDTO
import com.github.arhor.simple.expense.tracker.model.ExpenseResponseDTO
import com.github.arhor.simple.expense.tracker.util.TemporalRange
import java.time.LocalDate

interface ExpenseService {

    fun getUserExpenses(userId: Long, dateRange: TemporalRange<LocalDate>): List<ExpenseResponseDTO>

    fun getExpenseById(expenseId: Long, dateRange: TemporalRange<LocalDate>): ExpenseResponseDTO

    fun createUserExpense(userId: Long, requestDTO: ExpenseRequestDTO): ExpenseResponseDTO
}
